portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java

```diff
@@ -170,6 +181,44 @@ public void testFindByCompanyId() throws Exception {
 	}
 
 	@Test
+	public void testFindByCompanyIdByUserGroupGroup() throws Exception {
+		_userGroup = UserGroupTestUtil.addUserGroup();
+		_userGroupGroup = GroupTestUtil.addGroup();
+		_userGroupUser = UserTestUtil.addUser();
+
+		GroupLocalServiceUtil.addUserGroupGroup(
```

Comment:
Hey Pei-Jung,

I'm very confused by that answer. What you are doing in the test is not what is being done when the User Group is created from the UI, is it?

I'm not sure I understand why you need to change the classNameId. We can talk about it on Slack if you prefer.